librz/analysis/sign.c

```diff
@@ -126,10 +126,10 @@ RZ_API RzList *rz_sign_fcn_xrefs(RzAnalysis *a, RzAnalysisFunction *fcn) {
 	}
 
 	RzList *ret = rz_list_newf((RzListFree)free);
-	RzList *xrefs = rz_analysis_function_get_xrefs(fcn);
-	rz_list_foreach (xrefs, iter, refi) {
-		if (refi->type == RZ_ANALYSIS_REF_TYPE_CODE || refi->type == RZ_ANALYSIS_REF_TYPE_CALL) {
-			const char *flag = getRealRef(core, refi->addr);
+	RzList *xrefs = rz_analysis_function_get_xrefs_to(fcn);
+	rz_list_foreach (xrefs, iter, xrefi) {
```

Comment:
These functions like `rz_analysis_function_xrefs_to` internally access to the hashtable, then convert filtered output to the RzList. I wonder if maybe it's better to refactor this code to iterate only once through the hashtable itself. It will certainly increase performance and decrease the memory consumption in case of the big and complex binaries.

Reply:
Well sounds good but this is unrelated to my PR.